hs_composite_resource/models.py

```diff
@@ -440,65 +769,138 @@ def supports_rename_path(self, src_full_path, tgt_full_path):
             src_file_dir = src_full_path
 
         if src_ext and tgt_ext:
-            is_renaming_file = True
+            if src_file_name != tgt_file_name:
+                is_renaming_file = True
+            else:
+                is_moving_file = True
         elif src_ext:
             is_moving_file = True
         elif not istorage.exists(tgt_file_dir):
-            # renaming folder - no restriction
-            return True
+            src_base_dir = os.path.dirname(src_full_path)
+            tgt_base_dir = os.path.dirname(tgt_full_path)
+            if src_base_dir == tgt_base_dir:
+                # renaming folder - no restriction
+                return True
+            is_moving_folder = True
         else:
             is_moving_folder = True
 
-        def check_file_rename_or_move():
-            # see if the folder containing the file represents an aggregation
-            if src_file_dir != self.file_path:
-                aggregation_path = src_file_dir[len(self.file_path) + 1:]
-                try:
-                    aggregation = self.get_aggregation_by_name(aggregation_path)
-                    return aggregation.supports_resource_file_rename
-                except ObjectDoesNotExist:
-                    # check if the source file represents an aggregation
-                    # get source resource file object from source file path
-                    src_res_file = ResourceFile.get(self, src_file_name, aggregation_path)
-                    aggregation = src_res_file.logical_file
-                    if aggregation is None:
-                        raise ObjectDoesNotExist("No aggregation found at {}".format(
-                            aggregation_path))
-                    if is_renaming_file:
-                        return aggregation.supports_resource_file_rename
-                    else:
-                        return aggregation.supports_resource_file_move
-            else:
-                # get source resource file object from source file path
-                src_res_file = ResourceFile.get(self, src_file_name)
-                # check if the source file is part of an aggregation
-                aggregation = src_res_file.logical_file
-                if aggregation is None:
-                    raise ObjectDoesNotExist("No aggregation found at {}".format(src_file_name))
+        def check_target_folder(src_aggr=None):
```

Comment:
This is going to be hard to maintain and fix bugs.  I suggest making a method like this on the parent logical file class and call it from here.  The parent logical file method can either be extended or overwritten by children.  This will help separate type specific logic into their respective implementations as well as encourage reuse.